presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushAggregationThroughOuterJoin.java

```diff
@@ -140,6 +140,11 @@ public Result apply(AggregationNode aggregation, Captures captures, Context cont
                 aggregation.getHashSymbol(),
                 aggregation.getGroupIdSymbol());
 
+        ImmutableList.Builder<Symbol> outputSymbols = ImmutableList.<Symbol>builder()
+                .addAll(getOuterTable(join).getOutputSymbols());
+        aggregation.getHashSymbol().ifPresent(outputSymbols::add);
```

Comment:
If the aggregation has a hash symbol, it probably comes from the join's outer table (since aggregation is on outer table's outputs). But then, it's an additional output symbol and not grouping key of the aggregation, so the `groupsOnAllColumns` cannot be satisfied.

I think this rule doesn't have -- and doesn't have to have -- ambitions to handle hash symbols.
It's enough if it detects this situation and bails out.